packages/aws-cdk-lib/aws-opensearchservice/lib/domain.ts

```diff
@@ -440,6 +447,58 @@ export enum IpAddressType {
   DUAL_STACK = 'dualstack',
 }
 
+/**
+ * Configuration for a specific node type in OpenSearch domain
+ */
+export interface NodeConfig {
+  /**
+   * Whether this node type is enabled
+   *
+   * @default - false
+   */
+  readonly enabled?: boolean;
+
+  /**
+   * The instance type for the nodes
+   *
+   * @default - m5.large.search
+   */
+  readonly type?: string;
+
+  /**
+   * The number of nodes of this type
+   *
+   * @default - 1
+   */
+  readonly count?: number;
+}
+
+/**
+ * NodeType is a string enum of the node types in OpenSearch domain
+ *
+ */
+export enum NodeType {
+  /**
+   * Coordinator node type
+   */
+  COORDINATOR = 'coordinator',
+}
+
+/**
+ * Configuration for node options in OpenSearch domain
+ */
+export interface NodeOptions {
+  /**
+   * The type of node. Currently only 'coordinator' is supported.
+   */
+  readonly nodeType: NodeType;
+
+  /**
+   * Configuration for the node type
+   */
+  readonly nodeConfig: NodeConfig;
```

Comment:
Does this need to be a required property when the inner properties are all optional?

Reply:
Can we rely on the backend to respond to this one? Meaning right now we have a logic on what is required.

eg - if enabled = true, then we need to pass type and count, if enabled = false, type and count become optional.

Will keep this logic as it as is for now.

Reply:
Not sure how this is related to the backend. I'm mainly talking about the usage like below. Since the properties of `NodeConfig` are all optional, we can make `nodeConfig` an optional property as well so users are not limited to
```
{
  nodeType: someType,
  nodeConfig: {}
}
```
and can also simplify this to below
```
{
  nodeType: someType,
}
```